README.md

```diff
@@ -34,16 +34,16 @@ RAG is a method that tries to avoid hallucination by using information that is p
 
 ### How does RAG work?
 
-0. Context texts are loaded into the database. They are embedded so that it is easier to search for the most relevant information.
-1. A **user asks a question** (inputs a query)
-2. The query is embedded and most related **contexts are retrieved** from the database. The distance between the embedded query and the embedded context determines the similarity between the query and the contexts.
-3. The k most related contexts(in text) are combined with the user's query(in text) to **generate a better question** (query in text) to feed to an LLM. Better question: a question that includes the context and asks to say "I don't know" if the information is not present in the context.
-4. The query generated from the last step is fed to an LLM model, and the **LLM model gives the user an answer based on the query** generated from the last step. The user receives this answer.
+0. Embeddings of context texts are loaded into the database. The texts are embedded for the similarity search in later steps.
+1. A **user asks a question** (inputs a query).
+2. **k most related contexts to the query are retrieved** from the database. The distance between the embedded query and the embedded context determines the similarity between the query and the contexts.
+3. The k most related contexts(in text) are combined with the user's query(in text) to make a better prompt. It may simply be a **concatenation of the query and contexts with additional prompts**. Some works add prompts to respond "I don't know" if the information is not present in the context to minimize hallucinations.
+5. The better prompt made in the last step is fed to the LLM model, and the **LLM model gives the user an answer** based on the better prompt. The user receives this answer.
```

Comment:
excuse me for nitpicking too much today 😂 but I think this number should be "4" instead of "5", though it shows the number correctly on UI

Reply:
ahhhhhhh thank youuuuuuu
and no problem! I love your neat and precise review!